src/popup/components/Header.vue

```diff
@@ -85,60 +91,43 @@ export default {
     BtnPlain,
     BtnIcon,
   },
-  subscriptions() {
-    return {
-      superheroNotifications: this.$store.state.observables.notifications,
-    };
-  },
-  computed: {
-    ...mapGetters(['isLoggedIn']),
-    ...mapState(['notifications', 'pageTitle']),
-    truncateStr() {
-      const { pageTitle, title } = this;
-      return pageTitle || (title ? this.$t(`pages.titles.${title}`) : '');
-    },
-    title() {
-      return this.$route.meta.title;
-    },
-    isModalHeader() {
-      return this.$route.meta.isModalHeader;
-    },
-    showBack() {
-      return (
-        (this.$route.meta.backButton !== undefined ? this.$route.meta.backButton : true)
-        && this.title
-      );
-    },
-    showMore() {
-      return this.$route.name !== 'more' && !this.$route.meta?.closeButton;
-    },
-    showNotifications() {
-      return this.$route.name !== 'notifications' && !this.$route.meta?.hideNotificationsIcon;
-    },
-    isLogoDisabled() {
-      return this.$route.name === 'account';
-    },
-  },
-  methods: {
-    back() {
-      if (this.$store.state.route.from.path === '/') {
-        this.$router.push(this.isLoggedIn ? '/account' : '/');
-        return;
+  setup(props, { root }) {
+    const isLoggedIn = useGetter('isLoggedIn');
+    const currentHomeRouteName = computed(() => (isLoggedIn.value) ? ROUTE_ACCOUNT : ROUTE_INDEX);
+    const routeMeta = computed(() => root.$route.meta as WalletRouteMeta);
+    const showHeaderNavigation = computed(() => !!routeMeta.value?.showHeaderNavigation);
+    const isLogoDisabled = computed(() => root.$route.name === ROUTE_ACCOUNT);
+    const titleTruncated = computed(
+      () => routeMeta.value?.title ? root.$t(`pages.titles.${routeMeta.value.title}`) : '',
+    );
```

Comment:
```suggestion
    const titleTruncated = computed<string>(
      () => routeMeta.value?.title ? root.$t(`pages.titles.${routeMeta.value.title}`) : '',
    );
```

Reply:
Actually it's not string but `TranslateResult`. Forcing it to be a `<string>` creates an error. But in practice the type is inferred correctly.